Protocol.md

```diff
@@ -1,16 +1,37 @@
-# DLC Protocol
+# Peer Protocol for Contract Negotiation
 
-This document describes a protocol for two parties to set up a Discreet Log Contract (DLC).
+# Table of Contents
 
-## Preliminaries
+  * [Contract](#contract)
+      * [Definition of `contract_id`](#definition-of-contract_id)
+      * [Contract Negotiation](#contract-negotiation)
+          * [The `offer_dlc` Message](#the-offer_dlc-message)
+          * [The `accept_dlc` Message](#the-accept_dlc-message)
+          * [The `sign_dlc` Message](#the-sign_dlc-message)
+* [Authors](#authors)
 
-The following protocol assumes that both parties have already agreed on the contract terms (the possible outputs of the contract).
-Negotiation of the contract, or decisions on the underlying asset are thus not part of the exchanged information.
+# Contract
 
-## Description
+## Definition of `contract_id`
 
-The two parties exchange a set of three messages, and terminates when the non-initiating party broadcasts the Fund Transaction to the Bitcoin network.
-The following figure gives a brief overview of the protocol:
+Most messages use a `contract_id` to identify the contract. It's
+derived from the funding transaction by combining the `funding_txid`
+and the `funding_output_index`, using big-endian exclusive-OR
+(i.e. `funding_output_index` alters the last 2 bytes).
+
+Prior to a contract being accepted, a `temporary_contract_id` is used,
+which is the SHA256 hash of the offer message.
```

Comment:
This is a good point, perhaps we should include the `temporary_contract_id` in the `contract_id`? Or just part of it? Thoughts?

Reply:
Maybe just adding a nonce to the offer can fix this

Reply:
I guess both proposals are equivalent since IIUC `temporary_contract_id` is a random nonce? 

I would really like to have an ID that can be set only by the offerer, but I can't come up with something that really works properly so I guess reusing the temporary id is the best.